demo/esri-ds2022/src/components/label/label-content/label-content.tsx

```diff
@@ -104,11 +66,55 @@ export class LabelContent {
           this.labelClass.maxScale = value;
         }
         this.closeLabelPopovers.emit();
-        this.reRender = !this.reRender;
+        this.internalLabelUpdated.emit();
       }
     );
+    this.mapViewScaleWatch = this.mapView.watch("scale", () => {
+      this.scaleRangeSlider.renderNow();
+    });
   }
 
+  disconnectedCallback() {
+    this.scaleRangeSlider?.destroy();
+    this.scaleRangeSliderWatch?.remove();
+    this.mapViewScaleWatch?.remove();
+  }
+
+  @Listen("closeLabelPopovers", { target: "window" })
+  closeLabelPopoversHandler(): void {
+    if (this.labelStyle) {
+      document.body.removeChild(this.labelStyle);
+      this.labelStyle = null;
+    }
+    this.disableLabelPanel.emit(false);
+  }
+
+  getDisplayFieldName(): string {
+    return (this.labelClass as any).getLabelExpressionSingleField();
+  }
+
+  labelFieldSelection = (): void => {
+    this.closeLabelPopovers.emit();
+    let selectedItem = this.dropdownElement?.selectedItems?.[0]?.id;
+    this.dropdownButton.innerHTML = selectedItem;
+    this.labelClass.labelExpressionInfo.expression = `$feature["${selectedItem}"]`;
+    this.internalLabelUpdated.emit();
+  };
+
+  openLabelStyle = (): void => {
+    this.closeLabelPopovers.emit();
+    if (!this.labelStyle) {
+      this.labelStyle = document.createElement("esri-ds2022-label-content-style");
+      this.labelStyle.labelContentRefElement = this.hostElement;
+      this.labelStyle.labelClass = this.labelClass;
+      this.labelStyle.addEventListener("labelContentStyleChanges", () => {
```

Comment:
we should store this listener and remove it when necessary as well.

Reply:
I think it should be removed in `closeLabelPopoversHandler` and `disconnectedCallback`.